### PR TITLE
D3ASIM-2190: External grid_fee API enhancement

### PR DESCRIPTION
**Pull request by faizan2590**

### CODE REVIEW

File: src/d3a/models/area/redis_external_market_connection.py [hunk 1]
```diff
@@ -48,25 +48,43 @@ def market_stats_callback(self, payload):
...
+                self.area.config.grid_fee_type == 1:
+            self.area.transfer_fee_const = payload_data["fee_const"]
+            self.publish_json(grid_fees_response_channel, {
+                "status": "ready", "command": "grid_fees",
+                "market_fee_const": str(self.area.transfer_fee_const),
+                "transaction_id": payload_data.get("transaction_id", None)}
+             )
+        elif "fee_percent" in payload_data and payload_data["fee_percent"] is not None and \
+                self.area.config.grid_fee_type == 2:
+            self.area.grid_fee_percentage = payload_data["fee_const"]
```
Comment (spyrostz) [inline src/d3a/models/area/redis_external_market_connection.py:63]: I assume that this should be 'fee_percent', and not 'fee_const'

File: src/d3a/models/area/redis_external_market_connection.py [hunk 2]
```diff
@@ -48,25 +48,43 @@ def market_stats_callback(self, payload):
...
+        if "fee_const" in payload_data and payload_data["fee_const"] is not None and \
+                self.area.config.grid_fee_type == 1:
+            self.area.transfer_fee_const = payload_data["fee_const"]
+            self.publish_json(grid_fees_response_channel, {
+                "status": "ready", "command": "grid_fees",
+                "market_fee_const": str(self.area.transfer_fee_const),
+                "transaction_id": payload_data.get("transaction_id", None)}
+             )
+        elif "fee_percent" in payload_data and payload_data["fee_percent"] is not None and \
+                self.area.config.grid_fee_type == 2:
```
Comment (spyrostz) [inline src/d3a/models/area/redis_external_market_connection.py:62]: I am missing some validation here, and in the const branch. The incoming fees should be in the range [0-100] for the percentage fees, [0-200] for the constant fees.

**Comment by hannesdiedrich** [comment]
Once Spyros comments are considered, LGTM

File: src/d3a/models/area/redis_external_market_connection.py (hunk 2 above)
Comment (faizan2590) [reply src/d3a/models/area/redis_external_market_connection.py:65]: Yes using `d3a-interface` grid_fee_validator

**Pull request merged**